src/graph_notebook/notebooks/03-Sample-Applications/04-Security-Graphs/01-Building-a-Security-Graph-Application-with-Gremlin.ipynb

```diff
@@ -0,0 +1,580 @@
+    "\n",
+    "MATCH p=(policy:`iam:policy`)-[:resource_link]-(role:`iam:role`)-\n",
+    "[:resource_link|transient_resource_link*2]-(ec2:`ec2:instance`)\n",
+    "where (policy.name='AdministratorAccess' \n",
+    "OR policy.default_version_policy_document_text CONTAINS '*')\n",
+    "AND exists(ec2.public_ip_address)\n",
+    "RETURN p"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f472c0bc",
+   "metadata": {},
+   "source": [
+    "### How are they exposed?\n",
+    "\n",
+    "Now that we see that both of these instances are publicly accessible from the internet let's look for the associated security groups to see where we might want to start remedying this risk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a98d95ec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%gremlin\n",
+    "g.V().hasLabel('iam:policy').where(has('name', 'AdministratorAccess').or().\n",
+    "has('default_version_policy_document_text', containing('*'))).\n",
+    "bothE('resource_link').otherV().hasLabel('iam:role').\n",
+    "emit().\n",
+    "repeat(bothE('resource_link','transient_resource_link').otherV().simplePath()).times(2).\n",
+    "hasLabel('ec2:instance').has('public_ip_address').both('resource_link','transient_resource_link').\n",
+    "simplePath().hasLabel('ec2:security-group').\n",
+    "path().by(elementMap())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "df48d661",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%oc \n",
+    "\n",
+    "MATCH p=(policy:`iam:policy`)-[:resource_link]-(role:`iam:role`)-\n",
+    "[:resource_link|transient_resource_link*2]-(ec2:`ec2:instance`)\n",
+    "-[:resource_link|transient_resource_link]-(sg:`ec2:security-group`)\n",
+    "where (policy.name='AdministratorAccess' \n",
+    "OR policy.default_version_policy_document_text CONTAINS '*')\n",
+    "AND exists(ec2.public_ip_address)\n",
+    "RETURN p"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b5828b6",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "This notebook has shown how you can use Amazon Neptune to create a security graph as part of a cloud posture management solution. We've used a real-world AWS account graph to locate unused or potentially insecure resources within our infrastructure.  We then explored the connected resources to see the extent of the potential risk caused by these resources.  Once we had a feel for the potential risk we looked at the potential exploitation paths a bad actor may exploit and identified areas to look for a remedy.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b3cbe64",
+   "metadata": {},
+   "source": [
+    "## What's Next\n",
+    "\n",
+    "The examples in this notebook show how to develop a security graph data model and accompanying queries. To build a security graph solution that incorporates Neptune, we recommend the following resources:\n",
```

Comment:
Let's add the new security graphs on AWS page: https://aws.amazon.com/neptune/security-graphs-on-aws/

Reply:
fix "infrastructur"